### PR TITLE
Migrated Rescheduler to new taints

### LINKED ISSUE #2382: Make rescheduler work with Kubernetes 1.6
Taints are being moved from annotation to field of Node, and tolerations are being moved from annotation to field of PodSpec. See kubernetes/kubernetes#25320 and kubernetes/kubernetes#38957

There's no 1.6 milestone in this repo, but it is needed for 1.6.

### PR DESCRIPTION
**Pull request by piosz**
fix #2382
ref https://github.com/kubernetes/kubernetes/issues/32531

cc @ethernetdan  @marun @k82cn @aveshagarwal

### CODE REVIEW

File: rescheduler/rescheduler.go
Comment (aveshagarwal) [inline rescheduler/rescheduler.go]: why are you adding this annotation key?

Comment (piosz) [inline rescheduler/rescheduler.go]: It's needed for annotations cleanup in releaseAllTaintsDepracated function below.

File: rescheduler/rescheduler.go
Comment (k82cn) [inline rescheduler/rescheduler.go]: any const for it?

File: rescheduler/rescheduler.go
Comment (k82cn) [inline rescheduler/rescheduler.go]: No sure the relationship with CA, prefer not to dependent on it if possible. I'm ok with that for e2e.
For the common tools, I'd suggest to move to client-go.

**Review by k82cn — APPROVED** [review]

File: rescheduler/rescheduler.go
Comment (davidopp) [inline rescheduler/rescheduler.go]: stealing a comment @liggitt made on another PR -- it's probably better to make sure the taint does not already exist, and only add it if it does not already exist.

**Review by davidopp — COMMENTED** [review]
Just a couple of small comments

File: rescheduler/rescheduler.go
Comment (k82cn) [inline rescheduler/rescheduler.go]: Got that 👍

Comment (piosz) [inline rescheduler/rescheduler.go]: This was by design. Rescheduler shares logic with CA pretty much, because it performs similar simulations. The new dependency is just a result of refactoring in CA: some functions were move out of the simulator package to drain.

File: rescheduler/rescheduler.go
Comment (piosz) [inline rescheduler/rescheduler.go]: Done

File: rescheduler/rescheduler.go
Comment (piosz) [inline rescheduler/rescheduler.go]: Done

File: rescheduler/rescheduler.go
Comment (piosz) [inline rescheduler/rescheduler.go]: done

File: rescheduler/rescheduler.go
Comment (piosz) [inline rescheduler/rescheduler.go]: Ups. Thanks for the catch

File: rescheduler/rescheduler.go
Comment (piosz) [inline rescheduler/rescheduler.go]: It's already checked in function checkTaints, so there is no need to check it again here.

**Comment by piosz** [comment]
@davidopp comments addressed PTAL

**Comment by piosz** [comment]
Addressed all comments, the comments were minor, so merging manually without LGTM to unblock other work.

**Comment by marun** [comment]
@piosz I find it at a bit unnerving that you merged this yourself.  I get that there was urgency around its release-blocking status, but I think that makes oversight all the more critical.

Also, are there no unit or integration tests for this code?

**Comment by piosz** [comment]
@marun sorry for that. Working across multiple timezones has disadvantage of waiting for the next round of review for the whole day. Hopefully @davidopp will approve it soon.

I know that there is no unit tests (there is e2e test in core Kubernetes) and I plan to fix it later this week or early next week (see #2465).

**Pull request merged**
